app/controllers/backend/financial_years_controller.rb

```diff
@@ -90,9 +90,13 @@ def show
     def new
       @financial_year = FinancialYear.new
       f = FinancialYear.last
-      @financial_year.started_on = f.stopped_on + 1 unless f.nil?
+      @financial_year.started_on = f.nil? ? Entity.of_company&.born_at : f.stopped_on + 1
       @financial_year.started_on ||= Time.zone.today
-      @financial_year.stopped_on = ((@financial_year.started_on - 1) >> 12).end_of_month
+      if f.nil? && Entity.of_company&.first_financial_year_ends_on
```

Comment:
I don't get why we're checking on `f.nil?` here? It's not called in any of the clauses.